### ISSUE TITLE
MIssing CUDA kernel operator

### ISSUE THREAD

**Issue by paynowski** [pytorch/pytorch#30077]
## 🐛 Bug

I receive this error when trying to load a script model:
```
cuda_dispatch_ptr INTERNAL ASSERT FAILED at C:\w\1\s\windows\pytorch\aten\src\ATen/native/DispatchStub.h:74, please report a bug to PyTorch. DispatchStub: missing CUDA kernel (operator () at C:\w\1\s\windows\pytorch\aten\src\ATen/native/DispatchStub.h:74)
(no backtrace available)
error running the model
Method 'forward' is not defined. (get_method at C:\w\1\s\windows\pytorch\torch/csrc/jit/script/module.h:194)
(no backtrace available)
```

## To Reproduce

1. I export my UNet via tracing and scripting
2. When I try loading in C++ app I get the error message above. 

The 2 export methods

def ExportScriptViaTracing(self):
        """
        Converting Python model to Torch Script via Tracing for C++
        """

        self.Model.eval()
        example = torch.ones((self.__cfg.BATCH_SIZE, self.__cfg.IMG_CHANNELS, self.__cfg.IMG_HEIGHT, self.__cfg.IMG_WIDTH), dtype=torch.float32).to(self.Device)
        trace_module = torch.jit.trace(self.Model, example)
        trace_module.save('./{}/{}_{}.tm'.format(self.ModelDir, self.Name, self.Device.type))
        output1 = modelInfo.Model(example)
        output2 = trace_module(example)
        if torch.all(torch.eq(output1, output2)):
            logging.info("Verified tracing")

    def ExportScriptViaAnnotation(self):
        """
        Converting Python model to Torch Script via Annotation for C++
        """
        script_module = torch.jit.script(self.Model)
        script_module.save('./{}/{}_{}.sm'.format(self.ModelDir, self.Name, self.Device.type))

class UNet(nn.Module):

    def conv_block(self, in_channels, out_channels, act_fn):
        model = nn.Sequential(
            nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1, padding=1),
            nn.BatchNorm2d(out_channels),
            act_fn,
        )
        return model

    def conv_trans_block(self, in_channels, out_channels, act_fn):
        model = nn.Sequential(
            nn.ConvTranspose2d(in_channels, out_channels, kernel_size=3, stride=2, padding=1, output_padding=1),
            nn.BatchNorm2d(out_channels),
            act_fn,
        )
        return model

    def maxpool(self):
        pool = nn.MaxPool2d(kernel_size=2, stride=2, padding=0)
        return pool

    def conv_block_2(self, in_channels, out_channels, act_fn):
        model = nn.Sequential(
            self.conv_block(in_channels, out_channels, act_fn),
            nn.Conv2d(out_channels, out_channels, kernel_size=3, stride=1, padding=1),
            nn.BatchNorm2d(out_channels),
        )
        return model
    
    def __init__(self, in_dim:int, out_dim:int, num_filter:int=16):
        super().__init__()
        self.in_dim = in_dim
        self.out_dim = out_dim
        self.num_filter = num_filter
        act_fn = nn.LeakyReLU(0.2, inplace=True)

        # Encode
        self.down_1 = self.conv_block_2(self.in_dim, self.num_filter, act_fn) # 1, 16
        self.pool_1 = self.maxpool()
        self.down_2 = self.conv_block_2(self.num_filter*1, self.num_filter*2, act_fn) # 16, 32
        self.pool_2 = self.maxpool()
        self.down_3 = self.conv_block_2(self.num_filter*2, self.num_filter*4, act_fn) # 32, 64
        self.pool_3 = self.maxpool()
        self.down_4 = self.conv_block_2(self.num_filter*4, self.num_filter*8, act_fn) # 64, 128
        self.pool_4 = self.maxpool()

        self.bridge = self.conv_block_2(self.num_filter*8, self.num_filter*16, act_fn) # 128, 256

        # Decode
        self.trans_1 = self.conv_trans_block(self.num_filter*16,  self.num_filter*8, act_fn) # 256, 128
        self.up_1 = self.conv_block_2(self.num_filter*16, self.num_filter*8, act_fn)
        self.trans_2 = self.conv_trans_block(self.num_filter*8, self.num_filter*4, act_fn) # 128, 64
        self.up_2 = self.conv_block_2(self.num_filter*8, self.num_filter*4, act_fn)
        self.trans_3 = self.conv_trans_block(self.num_filter*4, self.num_filter*2, act_fn) # 64, 432
        self.up_3 = self.conv_block_2(self.num_filter*4, self.num_filter*2, act_fn)
        self.trans_4 = self.conv_trans_block(self.num_filter*2, self.num_filter*1, act_fn) # 32, 16
        self.up_4 = self.conv_block_2(self.num_filter*2, self.num_filter*1, act_fn)

        self.out = nn.Sequential(
            nn.Conv2d(self.num_filter, self.out_dim,  kernel_size=3, stride=1, padding=1), # 16, 1
            nn.Tanh(),
        )

    def forward(self, input):
        # Encode
        down_1 = self.down_1(input)
        pool_1 = self.pool_1(down_1)
        down_2 = self.down_2(pool_1)
        pool_2 = self.pool_2(down_2)
        down_3 = self.down_3(pool_2)
        pool_3 = self.pool_3(down_3)
        down_4 = self.down_4(pool_3)
        pool_4 = self.pool_4(down_4)

        bridge = self.bridge(pool_4)

        # Decode
        trans_1 = self.trans_1(bridge)
        concat_1 = torch.cat([trans_1, down_4], dim=1)
        up_1 = self.up_1(concat_1)
        trans_2 = self.trans_2(up_1)
        concat_2 = torch.cat([trans_2, down_3], dim=1)
        up_2 = self.up_2(concat_2)
        trans_3 = self.trans_3(up_2)
        concat_3 = torch.cat([trans_3, down_2], dim=1)
        up_3 = self.up_3(concat_3)
        trans_4 = self.trans_4(up_3)
        concat_4 = torch.cat([trans_4, down_1], dim=1)
        up_4 = self.up_4(concat_4)

        out = self.out(up_4)

        return out

## Expected behavior

Either model to load in the C++ application.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
Collecting environment information...
PyTorch version: 1.3.1
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: Quadro P4000
Nvidia driver version: Could not collect
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip] numpy==1.16.5

[pip] numpydoc==0.9.1

[pip] segmentation-models-pytorch==0.0.2

[pip] torch==1.3.1

[pip] torchsummary==1.5.1

[pip] torchvision==0.4.2
[conda] blas                      1.0                         mkl  

[conda] mkl                       2019.4                      245  

[conda] mkl-service               2.3.0            py37hb782905_0  

[conda] mkl_fft                   1.0.14           py37h14836fe_0  

[conda] mkl_random                1.1.0            py37h675688f_0  

[conda] pytorch                   1.3.1           py3.7_cuda101_cudnn7_0    pytorch

[conda] segmentation-models-pytorch 0.0.2                    pypi_0    pypi

[conda] torchsummary              1.5.1                    pypi_0    pypi

[conda] torchvision               0.4.2                py37_cu101    pytorch

## Additional context

cc @suo @ngimel

**Comment by peterjc123** [comment]
You'll need to try with the nightly build.

**Comment by paynowski** [comment]
I've just tried the latest nightly build. Now I get linker errors.

 Creating library M:/code/DL-Experiments/UNet-PyTorch/TorchScriptModel/Example/Debug/Example.lib and object M:/code/DL-Experiments/UNet-PyTorch/TorchScriptModel/Example/Debug/Example.exp
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) private: __cdecl caffe2::TypeIdentifier::TypeIdentifier(unsigned short)" (__imp_??0TypeIdentifier@caffe2@@AEAA@G@Z) referenced in function "public: static class caffe2::TypeIdentifier __cdecl caffe2::TypeIdentifier::Get<signed char>(void)" (??$Get@C@TypeIdentifier@caffe2@@SA?AV01@XZ)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: struct c10::TensorOptions __cdecl c10::TensorOptions::is_variable(class c10::optional<bool>)const " (__imp_?is_variable@TensorOptions@c10@@QEBA?AU12@V?$optional@_N@2@@Z) referenced in function "public: class at::Tensor __cdecl <lambda_3e994ac619644f7a071f6bbca0c90991>::operator()(void)const " (??R<lambda_3e994ac619644f7a071f6bbca0c90991>@@QEBA?AVTensor@at@@XZ)
2>  Hint on symbols that are defined and could potentially match:
2>    "__declspec(dllimport) public: bool __cdecl at::Tensor::is_variable(void)const " (__imp_?is_variable@Tensor@at@@QEBA_NXZ)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) class c10::TensorTypeSet __cdecl c10::impl::tls_excluded_tensor_type_set(void)" (__imp_?tls_excluded_tensor_type_set@impl@c10@@YA?AVTensorTypeSet@2@XZ) referenced in function "enum c10::TensorTypeId __cdecl at::impl::dispatchTypeId(class c10::TensorTypeSet)" (?dispatchTypeId@impl@at@@YA?AW4TensorTypeId@c10@@VTensorTypeSet@4@@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) private: void __cdecl at::ATenOpTable::reportError(enum c10::TensorTypeId)const " (__imp_?reportError@ATenOpTable@at@@AEBAXW4TensorTypeId@c10@@@Z) referenced in function "public: class at::Tensor __cdecl at::ATenOpTable::callUnboxed<class at::Tensor,class c10::ArrayRef<__int64>,struct c10::TensorOptions const &>(class c10::ArrayRef<__int64>,struct c10::TensorOptions const &)const " (??$callUnboxed@VTensor@at@@V?$ArrayRef@_J@c10@@AEBUTensorOptions@4@@ATenOpTable@at@@QEBA?AVTensor@1@V?$ArrayRef@_J@c10@@AEBUTensorOptions@4@@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class at::ATenOpTable const * __cdecl at::ATenDispatch::getOpTable(char const *)const " (__imp_?getOpTable@ATenDispatch@at@@QEBAPEBVATenOpTable@2@PEBD@Z) referenced in function "class at::Tensor __cdecl at::ones(class c10::ArrayRef<__int64>,struct c10::TensorOptions const &)" (?ones@at@@YA?AVTensor@1@V?$ArrayRef@_J@c10@@AEBUTensorOptions@4@@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) class at::ATenDispatch & __cdecl at::globalATenDispatch(void)" (__imp_?globalATenDispatch@at@@YAAEAVATenDispatch@1@XZ) referenced in function "class at::Tensor __cdecl at::ones(class c10::ArrayRef<__int64>,struct c10::TensorOptions const &)" (?ones@at@@YA?AVTensor@1@V?$ArrayRef@_J@c10@@AEBUTensorOptions@4@@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl torch::autograd::Variable::Variable(void)" (__imp_??0Variable@autograd@torch@@QEAA@XZ) referenced in function "struct torch::autograd::Variable __cdecl torch::autograd::make_variable(class at::Tensor,bool,bool)" (?make_variable@autograd@torch@@YA?AUVariable@12@VTensor@at@@_N1@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) private: __cdecl torch::autograd::Variable::Variable(class c10::intrusive_ptr<struct c10::TensorImpl,struct c10::detail::intrusive_target_default_null_type<struct c10::TensorImpl> >)" (__imp_??0Variable@autograd@torch@@AEAA@V?$intrusive_ptr@UTensorImpl@c10@@U?$intrusive_target_default_null_type@UTensorImpl@c10@@@detail@2@@c10@@@Z) referenced in function "struct torch::autograd::Variable __cdecl torch::autograd::make_variable(class at::Tensor,bool,bool)" (?make_variable@autograd@torch@@YA?AUVariable@12@VTensor@at@@_N1@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl torch::autograd::Variable::~Variable(void)" (__imp_??1Variable@autograd@torch@@QEAA@XZ) referenced in function "int `struct autograd::make_variable::Variable __cdecl torch::autograd::make_variable(class at::Tensor,bool,bool)'::`1'::dtor$8" (?dtor$8@?0??make_variable@autograd@torch@@YA?AUVariable@12@VTensor@at@@_N1@Z@4HA)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl torch::autograd::Variable::AutogradMeta::AutogradMeta(struct c10::TensorImpl *,bool,struct torch::autograd::Edge)" (__imp_??0AutogradMeta@Variable@autograd@torch@@QEAA@PEAUTensorImpl@c10@@_NUEdge@23@@Z) referenced in function "class std::unique_ptr<struct torch::autograd::Variable::AutogradMeta,struct std::default_delete<struct torch::autograd::Variable::AutogradMeta> > __cdecl std::make_unique<struct torch::autograd::Variable::AutogradMeta,struct c10::TensorImpl *,bool &,0>(struct c10::TensorImpl * &&,bool &)" (??$make_unique@UAutogradMeta@Variable@autograd@torch@@PEAUTensorImpl@c10@@AEA_N$0A@@std@@YA?AV?$unique_ptr@UAutogradMeta@Variable@autograd@torch@@U?$default_delete@UAutogradMeta@Variable@autograd@torch@@@std@@@0@$$QEAPEAUTensorImpl@c10@@AEA_N@Z)
2>Example.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl torch::autograd::Variable::AutogradMeta::~AutogradMeta(void)" (__imp_??1AutogradMeta@Variable@autograd@torch@@UEAA@XZ) referenced in function "public: virtual void * __cdecl torch::autograd::Variable::AutogradMeta::`scalar deleting destructor'(unsigned int)" (??_GAutogradMeta@Variable@autograd@torch@@UEAAPEAXI@Z)
2>Example.obj : error LNK2001: unresolved external symbol "public: virtual struct torch::autograd::Variable & __cdecl torch::autograd::Variable::AutogradMeta::grad(void)" (?grad@AutogradMeta@Variable@autograd@torch@@UEAAAEAU234@XZ)
2>Example.obj : error LNK2001: unresolved external symbol "public: virtual struct torch::autograd::Variable const & __cdecl torch::autograd::Variable::AutogradMeta::grad(void)const " (?grad@AutogradMeta@Variable@autograd@torch@@UEBAAEBU234@XZ)
2>Example.obj : error LNK2001: unresolved external symbol "public: virtual bool __cdecl torch::autograd::Variable::AutogradMeta::requires_grad(void)const " (?requires_grad@AutogradMeta@Variable@autograd@torch@@UEBA_NXZ)
2>Example.obj : error LNK2001: unresolved external symbol "public: virtual void __cdecl torch::autograd::Variable::AutogradMeta::set_requires_grad(bool,struct c10::TensorImpl *)" (?set_requires_grad@AutogradMeta@Variable@autograd@torch@@UEAAX_NPEAUTensorImpl@c10@@@Z)
2>M:\code\DL-Experiments\UNet-PyTorch\TorchScriptModel\Example\Debug\Example.exe : fatal error LNK1120: 15 unresolved externals
2>Done building project "Example.vcxproj" -- FAILED.
========== Rebuild All: 1 succeeded, 1 failed, 0 skipped ==========

**Comment by peterjc123** [comment]
Please follow the tutorial [here](https://pytorch.org/tutorials/advanced/cpp_frontend.html). Please don't create a new VS project by yourself. Actually you can make use of the CMake scripts.

**Comment by paynowski** [comment]
I guess I have a more general deployment question. We are planning on incorporating different models into our product (PyTorch/Tensorflow). The product is C# based with some C++/CLI and C++ only where absolutely necessary.

Seems like the best routes are:
a) ONNX
b) C++ APIs

For PyTorch is the C++ Frontend better than TorchScript?

**Comment by peterjc123** [comment]
cc @ezyang

**Comment by ezyang** [comment]
@paynowski This is not really enough detail to answer your question, but I certainly wouldn't discard TorchScript, it has a different function than the C++ API. If you have researchers writing PyTorch models in Python, almost certainly TorchScript way is the right way to ship them. You could ask for more opinions about this on the forums.

**Comment by zener90818** [comment]
same error, Did you solve it?

**Comment by paynowski** [comment]
We dropped it at that point. We need this for a .NET deployed product and not research. The deployment options are still limited with PyTorch, unfortunately, and Tensforflow is better for deployment in our situation. Too bad, I really like PyTorch as a model creation environment.

 We chose ONNX exported models. We use Tensorflow Serving (server) and ML.NET (in process)
